main.c

```diff
@@ -528,40 +534,64 @@ CloseApplication(HWND hwnd)
 }
 
 void
-ImportConfigFile()
+ShowConfigFileDialog(PTCHAR config_file)
 {
-    
-    TCHAR filter[2*_countof(o.ext_string)+5];
+    PTCHAR fileName;
+    int dlgresult = 0;
+
+    fileName = PathFindFileName(config_file);
+    dlgresult = ShowLocalizedMsgEx(MB_YESNO, _T(PACKAGE_NAME), IDS_ASK_IMPORT_OR_EDIT, fileName);
+    if (dlgresult == IDYES)
+        ImportConfigFile(config_file);
+    else
+        EditConfig(0, config_file);
+}
 
-    _sntprintf_0(filter, _T("*.%s%c*.%s%c"), o.ext_string, _T('\0'), o.ext_string, _T('\0'));
-    
-    OPENFILENAME fn;
+
+void
+ImportConfigFile(PTCHAR config_file)
+{
     TCHAR source[MAX_PATH] = _T("");
+    PTCHAR fileName;
+    BOOL result = 1;
 
-    fn.lStructSize = sizeof(OPENFILENAME);
-    fn.hwndOwner = NULL;
-    fn.lpstrFilter = filter;
-    fn.lpstrCustomFilter = NULL;
-    fn.nFilterIndex = 1;
-    fn.lpstrFile = source;
-    fn.nMaxFile = MAX_PATH;
-    fn.lpstrFileTitle = NULL;
-    fn.lpstrInitialDir = NULL;
-    fn.lpstrTitle = NULL;
-    fn.Flags = OFN_DONTADDTORECENT | OFN_FILEMUSTEXIST;
-    fn.lpstrDefExt = NULL;
-
-    if (GetOpenFileName(&fn)) 
+    if (config_file == NULL)
+    {
+        OPENFILENAME fn;
+        TCHAR filter[2*_countof(o.ext_string)+5];
+
+        _sntprintf_0(filter, _T("*.%s%c*.%s%c"), o.ext_string, _T('\0'), o.ext_string, _T('\0'));
+
+        fn.lStructSize = sizeof(OPENFILENAME);
+        fn.hwndOwner = NULL;
+        fn.lpstrFilter = filter;
+        fn.lpstrCustomFilter = NULL;
+        fn.nFilterIndex = 1;
+        fn.lpstrFile = source;
+        fn.nMaxFile = MAX_PATH;
+        fn.lpstrFileTitle = NULL;
+        fn.lpstrInitialDir = NULL;
+        fn.lpstrTitle = NULL;
+        fn.Flags = OFN_DONTADDTORECENT | OFN_FILEMUSTEXIST;
+        fn.lpstrDefExt = NULL;
+
+        result = GetOpenFileName(&fn) && (fileName = source + fn.nFileOffset);
+    }
+    else
     {
+        fileName = PathFindFileName(config_file);
```

Comment:
We should check that config_file exists -- its not guaranteed to be a valid file as it comes from a command line. fileName is used below to create a subdir, so we want it to be a sane name.